src/rgw/rgw_rados.cc

```diff
@@ -2238,11 +2238,11 @@ int RGWRados::create_bucket(const RGWUserInfo& owner, rgw_bucket& bucket,
     info.placement_rule = selected_placement_rule;
     info.swift_ver_location = swift_ver_location;
     info.swift_versioning = (!swift_ver_location.empty());
-    init_default_bucket_layout(cct, info, svc.zone->get_zone());
-    if (pmaster_num_shards) {
-      // TODO: remove this once sync doesn't require the same shard count
-      info.layout.current_index.layout.normal.num_shards = *pmaster_num_shards;
-    }
+    // TODO: remove special case for pmaster_num_shards once sync doesn't require the same shard count
+    init_default_bucket_layout(cct, info, svc.zone->get_zone(),
+			       pmaster_num_shards ?
+			       std::optional<uint32_t>{*pmaster_num_shards} :
+			       std::nullopt);
```

Comment:
there's some duplicated initialization below! it will push another copy of the log layout, and this assignment:
```c++
info.layout.current_index.layout.type = rule_info.index_type;
```
will modify the index layout after `init_default_bucket_layout()` copies it into the log layout

Reply:
i addressed this stuff in https://github.com/cbodley/ceph/commit/935cc09e0cc4996476c38389a9ba0c0a5bc0eff9 on the branch https://github.com/cbodley/ceph/commits/wip-rgw-init-default-bucket-layout